models.py

```diff
@@ -347,19 +347,21 @@ def create(self, new_article: Article) -> ArticleResult:
             raise SystemError(f"Create new article DB: Error, {err}")
 
     def update(self, article_update: ArticleUpdate) -> bool:
-        keys = ('article_id', 'name', 'text', 'date')
-        sql = f"SELECT name, article_text, pub_date FROM articles\
-                WHERE id = '{article_update['id']}';"
+        keys = 'article_id', 'name', 'text', 'date'
+        sql = f"SELECT id, name, article_text, pub_date FROM articles\
+                WHERE id = {article_update.article_id};"
         try:
             self.cur.execute(sql)
             values = self.cur.fetchall()
             values = values[0]
             dict_from_db = dict(zip(keys, values))
-            dict_for_db = {**dict_from_db, **article_update}
+            dict_article = asdict(article_update)
+            dict_article_update = {k: v for k, v in dict_article.items() if v is not None}
+            dict_for_db = {**dict_from_db, **dict_article_update}
```

Comment:
Why do we need all this dict variables? Why do you use this and not using just data class object?